src/ic-cdk/src/api/call.rs

```diff
@@ -251,58 +297,25 @@ pub fn call_raw128(
     args_raw: &[u8],
     payment: u128,
 ) -> impl Future<Output = CallResult<Vec<u8>>> + Send + Sync {
-    call_raw_internal(id, method, args_raw, move || {
-        add_payment(payment);
-    })
+    call_raw_internal(id, method, args_raw, 0, payment)
 }
 
 fn call_raw_internal(
     id: Principal,
     method: &str,
     args_raw: &[u8],
-    payment_func: impl FnOnce(),
+    payment: u64,
+    payment128: u128,
 ) -> impl Future<Output = CallResult<Vec<u8>>> + Send + Sync {
-    let callee = id.as_slice();
     let state = Arc::new(RwLock::new(CallFutureState {
         result: None,
         waker: None,
+        id,
+        method: method.to_string(),
+        arg: args_raw.to_vec(),
```

Comment:
This introduces the overhead of a copy, and the arguments may be arbitrarily large. If it can be stored borrowed, it should be; otherwise, the data should be stored in Vec form to begin with, including changing the API if necessary.

Reply:
I've changed the type of `args_raw` to `Vec<u8>` so that the content can be moved. Are you ok with copying the method name? (I guess a method name is typically small and can never be larger than [20k](https://ic-interface-spec.netlify.app/#system-api-module).)

Reply:
I'd feel *slightly* better if the outer functions took `impl Into<String>` but it's not nearly as important as the argument.